From: Ryan Lee <ryan.lee@canonical.com>
Date: Wed, 1 Apr 2026 10:29:56 -0700
Subject: [PATCH] profiles: add /sys/kernel/mm/transparent_hugepage/enabled to
 base abstraction

The problem was originally reported against pollinate, but there are many
other programs that might want this check too.

Fixes: https://bugs.launchpad.net/bugs/2147031

Signed-off-by: Ryan Lee <ryan.lee@canonical.com>
---
 profiles/apparmor.d/abstractions/base | 1 +
 1 file changed, 1 insertion(+)

diff --git a/profiles/apparmor.d/abstractions/base b/profiles/apparmor.d/abstractions/base
index 82d45cc..cab28a3 100644
--- a/profiles/apparmor.d/abstractions/base
+++ b/profiles/apparmor.d/abstractions/base
@@ -111,6 +111,7 @@
   @{sys}/devices/system/cpu/possible r,
 
   # transparent hugepage support
+  @{sys}/kernel/mm/transparent_hugepage/enabled r,
   @{sys}/kernel/mm/transparent_hugepage/hpage_pmd_size r,
 
   # glibc's *printf protections read the maps file
